cdc/kv/region_cache.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kv
+
+import (
+	"sync"
+
+	"github.com/tikv/client-go/v2/tikv"
+	pd "github.com/tikv/pd/client"
+)
+
+var (
+	regionCacheInstance *tikv.RegionCache
+	regionCacheOnce     sync.Once
+)
+
+func getRegionCacheInstance(pd pd.Client) *tikv.RegionCache {
+	regionCacheOnce.Do(func() {
+		regionCacheInstance = tikv.NewRegionCache(pd)
+	})
```

Comment:
kv client would close region cache after it exits, which is not expected after using a singleton